--- termite-16.6/subprojects/vte/src/vte.cc	2023-07-02 22:57:30.000000000 +0200
+++ termite-16.6-clipboard/subprojects/vte/src/vte.cc	2023-08-24 02:47:16.876596909 +0200
@@ -6532,7 +6532,7 @@
 		/* Copy only if something was selected. */
                 if (!m_selection_resolved.empty() &&
 		    m_selecting_had_delta) {
-                        widget_copy(vte::platform::ClipboardType::PRIMARY,
+                        widget_copy(vte::platform::ClipboardType::CLIPBOARD,
                                     vte::platform::ClipboardFormat::TEXT);
 			emit_selection_changed();
 		}
@@ -10050,7 +10050,7 @@
 	m_selecting_had_delta = true;
         m_selection_resolved.set ({ start_row, start_col },
                                   { end_row, end_col });
-        widget_copy(vte::platform::ClipboardType::PRIMARY,
+        widget_copy(vte::platform::ClipboardType::CLIPBOARD,
                     vte::platform::ClipboardFormat::TEXT);
 	emit_selection_changed();
 
